src/App.tsx

```diff
@@ -8,7 +8,79 @@ import { TodoFilter } from './components/TodoFilter';
 import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
 
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+
+enum SelectedStatus {
+  All = 'all',
+  Active = 'active',
+  Completed = 'completed',
+}
+
+function isIncludesQuery(todoTitle: string, query: string) {
+  return todoTitle.toLowerCase().includes(query.toLowerCase());
+}
+
 export const App: React.FC = () => {
+  const [initialTodos, setInitialTodos] = useState<Todo[]>([]);
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [query, setQuery] = useState<string>('');
+  const [selectedTodo, setSelectedTodo] = useState<Todo | null>(null);
+  const [selectedTodoStatus, setSelectedTodoStatus] = useState<string>('all');
+
+  const handleSelectTodo = (todoId: number | null) => (
+    todoId
+      ? (setSelectedTodo(todos.find(todo => todo.id === todoId) || null))
+      : setSelectedTodo(null)
+  );
+
+  const handleSelectStatus = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedTodoStatus(event.target.value);
+  };
+
+  const handleQueryChange = (value: string) => {
+    setQuery(value);
+  };
+
+  useEffect(() => {
+    const todosFromServer = async () => {
+      const data = await getTodos();
+
+      setInitialTodos(data);
+    };
+
+    todosFromServer();
+  }, []);
+
+  useEffect(() => {
```

Comment:
in this case you can call only `setTodos` and pass here function helper with params... So create helper function avoiding duplicate `initialTodos.filter(({ title }) =>` (use switch case in helper function)